### PR TITLE
Changing version AWSSDK V1 to V2

### PR DESCRIPTION
**Pull request by shaavanga**
### Description
[Describe what this change achieves]
Updating the AWSSDK version1 to version2
 
### Issues Resolved
Contribute to #4017

 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -91,6 +91,10 @@ subprojects {
         implementation platform('com.fasterxml.jackson:jackson-bom:2.15.3')
         implementation platform('org.eclipse.jetty:jetty-bom:9.4.53.v20231009')
         implementation platform('io.micrometer:micrometer-bom:1.10.5')
+        implementation platform('software.amazon.awssdk:bom:2.17.17')
```
Comment (dlvenable) [inline build.gradle:94]: You shouldn't need these four lines. 

We already get the BOM here:

https://github.com/opensearch-project/data-prepper/blob/d501af69c10b93efbb0c68abeeda4f5023323e18/build.gradle#L251

And for the following three dependencies, we don't need to choose which dependencies to pull it at the root project. Let the projects pull in what they need.

So you should be able to delete all four of these lines and make no changes to the root Gradle project.

File: data-prepper-plugins/http-sink/build.gradle [hunk 2]
```diff
@@ -26,7 +26,7 @@ dependencies {
     implementation 'com.github.scribejava:scribejava-core:8.3.3'
     implementation project(path: ':data-prepper-core')
     implementation project(':data-prepper-plugins:parse-json-processor')
-    implementation 'com.amazonaws:aws-java-sdk-sts:1.12.395'
+    implementation 'software.amazon.awssdk:sts:2.21.23'
```
Comment (dlvenable) [inline data-prepper-plugins/http-sink/build.gradle:29]: The BOM defines the version. So you can change this to:

```
implementation 'software.amazon.awssdk:sts'
```

File: data-prepper-plugins/translate-processor/build.gradle [hunk 3]
```diff
@@ -11,8 +11,8 @@ plugins {
 dependencies {
     implementation project(':data-prepper-api')
     implementation 'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml'
-    implementation 'com.amazonaws:aws-java-sdk-s3:1.12.395'
-    implementation 'com.amazonaws:aws-java-sdk-sts:1.12.395'
+    implementation 'software.amazon.awssdk:s3:2.21.23'
```
Comment (dlvenable) [inline data-prepper-plugins/translate-processor/build.gradle:14]: The BOM in the parent defines the version. So you can have:

```
implementation 'software.amazon.awssdk:s3'
implementation 'software.amazon.awssdk:sts'
```

**Review by dlvenable — CHANGES_REQUESTED** [review]
Thanks @shaavanga for this contribution. I have a few comments on the build.gradle for some changes you should make. We'll want to verify that the build succeeds as well.

File: build.gradle (hunk 1 above)
Comment (shaavanga) [reply build.gradle:94]: I had updated the changes as suggested

File: data-prepper-plugins/http-sink/build.gradle (hunk 2 above)
Comment (shaavanga) [reply data-prepper-plugins/http-sink/build.gradle]: removed particular version and updated

**Review by asifsmohammed — APPROVED** [review]

**Review by oeyh — APPROVED** [review]

**Review by asifsmohammed — APPROVED** [review]

**Review by oeyh — APPROVED** [review]

**Review by dlvenable — APPROVED** [review]
Thank you @shaavanga !

**Pull request merged**